packages/next/build/webpack-config.js

```diff
@@ -183,6 +183,7 @@ export default async function getBaseWebpackConfig (dir, {dev = false, isServer
       ...nodePathList // Support for NODE_PATH environment variable
     ],
     alias: {
+      webpack: path.join(NEXT_PROJECT_ROOT_DIST, 'compiled', 'webpack'), // This doesn't appear to work
```

Comment:
You should use something like: https://github.com/ilearnio/module-alias but a simplified version (require hook). It should be registered at the top level so that for example `next.config.js` is loaded after the registration, this way we won't break next.config.js users that `require('webpack')` 🙏 

Adding the alias here will only ensure that when `webpack` is required inside pages (which you generally never want to anyway) is resolved